src/src/PluginStructs/P148_data_struct.cpp

```diff
@@ -5,23 +5,75 @@
 
 # define TM1621_PULSE_WIDTH   10   // microseconds (Sonoff = 100)
 
-// Commands sent during init
-# define TM1621_SYS_EN        0x01 // 0b00000001
-# define TM1621_LCD_ON        0x03 // 0b00000011
-# define TM1621_TIMER_DIS     0x04 // 0b00000100
-# define TM1621_WDT_DIS       0x05 // 0b00000101
-# define TM1621_TONE_OFF      0x08 // 0b00001000
+// Commands
+# define TM1621_SYS_DIS       0x00 // 0b00000000 = Turn off system oscillator and LCD bias generator
+# define TM1621_SYS_EN        0x01 // 0b00000001 = Turn on the system oscillator
+# define TM1621_LCD_OFF       0x02 // 0b00000010 = disables the LCD bias generator
+# define TM1621_LCD_ON        0x03 // 0b00000011 = enables the LCD bias generator
+# define TM1621_TIMER_DIS     0x04 // 0b00000100 = disable timer
+# define TM1621_WDT_DIS       0x05 // 0b00000101 = disable WDT
+# define TM1621_TIMER_EN      0x06 // 0b00000110 = enable timer
+# define TM1621_WDT_EN        0x07 // 0b00000111 = enable WDT
+# define TM1621_TONE_OFF      0x08 // 0b00001000 = turn off sound output
+# define TM1621_TONE_ON       0x09 // 0b00001001 = turn on sound output
+# define TM1621_CLR_TIMER     0x0C // 0b000011XX = clear timer
+# define TM1621_CLR_WDT       0x0E // 0b0000111X = clear WDT status
+
 # define TM1621_BIAS          0x29 // 0b00101001 = LCD 1/3 bias 4 commons option
+                                   // 0b0010abXc =>
+                                   // c=0: optional 1/2 bias
+                                   // c=1: optional 1/3 bias
+                                   // ab=00: Optional 2 segment common ports
+                                   // ab=01: Optional 3 segment common ports
+                                   // ab=10: optional 4 segment common ports
+                                   // N.B. 1/3 bias only possible with 1:3 and 1:4 mux ratio (segment common ports)
 # define TM1621_IRQ_DIS       0x80 // 0b100x0xxx
 
 // FIXME TD-er: Must make the contrast/bias configurable
+// Commands sent during init.
 const uint8_t tm1621_commands[] =
 { TM1621_SYS_EN, TM1621_LCD_ON, TM1621_BIAS, TM1621_TIMER_DIS, TM1621_WDT_DIS, TM1621_TONE_OFF, TM1621_IRQ_DIS };
 
+// Uncrustify may mangle the readability of the fonts
+//
+// *INDENT-OFF*
+
 // TM1621 Font
 //                                          0     1     2     3     4     5     6     7     8     9     -     off
 const uint8_t tm1621_digit_row[2][12] = { { 0x5F, 0x50, 0x3D, 0x79, 0x72, 0x6B, 0x6F, 0x51, 0x7F, 0x7B, 0x20, 0x00 },
-  { 0xF5, 0x05, 0xB6, 0x97, 0x47, 0xD3, 0xF3, 0x85, 0xF7, 0xD7, 0x02, 0x00 } };
+                                          { 0xF5, 0x05, 0xB6, 0x97, 0x47, 0xD3, 0xF3, 0x85, 0xF7, 0xD7, 0x02, 0x00 } };
+//                                          'A'   'b'   'C'   'b'   'E'   'F'   'G'   'H'   'i'   'J'   'K'   'L'   'M'   'n'   'o'   'P'   'q'   'r'   'S'   't'   'U'   'v'   'W'   'X'   'Y'   'Z'   '?'   ' '
```

Comment:
> 'b'   'C'   'b'   'E' 

Should probably be `'b'   'C'   'd'   'E' `?

Reply:
Yep :)